### PR TITLE
feat: add support for arxiv identifier in ArxivAPIWrapper()

### PR DESCRIPTION
**Pull request by LMC117**
- Description: this PR adds the support for arxiv identifier of the ArxivAPIWrapper. I modified the `run()` and `load()` functions in `arxiv.py`, using regex to recognize if the query is in the form of arxiv identifier (see [https://info.arxiv.org/help/find/index.html](https://info.arxiv.org/help/find/index.html)). If so, it will directly search the paper corresponding to the arxiv identifier. I also modified and added tests in `test_arxiv.py`.
  - Issue: #9047 
  - Dependencies: N/A
  - Tag maintainer: N/A

### CODE REVIEW

**Comment by baskaryan** [comment]
lgtm, cc @leo-gan

**Comment by leo-gan** [comment]
I've tried
`    loader = ArxivLoader(query="1605.08386 2308.07912 2308.07910", load_max_docs=3)`
and it returns all 3 papers. 
Disclaimer: if IDs are separated by `,` it didn't work! Only the space separator works.
 
I don't think we need special treatment for multiple paper IDs because it works right now.

**Comment by LMC117** [comment]
Hi, 

If the version number is specified in the query (e.g. 2212.00794v2), no results will be returned. So I think there is a need to handle the arxiv identifier separately.

**Comment by leo-gan** [comment]
> If the version number is specified in the query (e.g. 2212.00794v2), no results will be returned. So I think there is a need to handle the arxiv identifier separately.

OK. Then, please add unit tests to work with Ids.

**Comment by LMC117** [comment]
Hi @leo-gan, I've committed new unit tests. You can check that

**Review by leo-gan — APPROVED** [review]
Thank you! 
LGTM

File: libs/langchain/langchain/utilities/arxiv.py [hunk 1]
```diff
@@ -54,6 +58,14 @@ class ArxivAPIWrapper(BaseModel):
     load_all_available_meta: bool = False
     doc_content_chars_max: Optional[int] = 4000
 
+    def is_arxiv_identifier(self, query: str) -> bool:
```
[the remark this answers is not in the document]
Comment (LMC117) [reply libs/langchain/langchain/utilities/arxiv.py:61]: sure, I will add it soon

[the remark this answers is not in the document]
Comment (leo-gan) [reply libs/langchain/langchain/utilities/arxiv.py]: @LMC117 Looks great! LGTM

**Comment by baskaryan** [comment]
thanks @LMC117!

**Comment by leo-gan** [comment]
@LMC117 Hi , could you, please, resolve the merging issues? After that ping me and I push this PR for the review. Thanks!

**Review by leo-gan — APPROVED** [review]

**Comment by LMC117** [comment]
@leo-gan hi I resolved that the merging issue

File: libs/langchain/langchain/utilities/arxiv.py [hunk 2]
```diff
@@ -88,9 +104,15 @@ def run(self, query: str) -> str:
             query: a plaintext search query
         """  # noqa: E501
         try:
-            results = self.arxiv_search(  # type: ignore
-                query[: self.ARXIV_MAX_QUERY_LENGTH], max_results=self.top_k_results
-            ).results()
+            if self.is_arxiv_identifier(query):
+                results = self.arxiv_search(
+                    id_list=query[: self.ARXIV_MAX_QUERY_LENGTH].split(),
```
Comment (hwchase17) [inline libs/langchain/langchain/utilities/arxiv.py:109]: i think we dont want to do `self.ARXIV_MAX_QUERY_LENGTH` right?

**Pull request merged**